### PR TITLE
server: fix some hostNet handling in container creation

### PR DESCRIPTION
**Pull request by haircommander**
#### What type of PR is this?

/kind bug
#### What this PR does / why we need it:
as well as fix an issue where hostNet containers couldn't access cgroup heierarchy 
#### Which issue(s) this PR fixes:

fixes https://github.com/kubernetes/kubernetes/issues/104034
#### Does this PR introduce a user-facing change?

```release-note
Fix a bug where host network containers couldn't read their cgroup hierarchy 
```

### CODE REVIEW

**Comment by TomSweeneyRedHat** [comment]
LGTM
assuming happy tests

**Review by saschagrunert — COMMENTED** [review]
Code LGTM, having an integration test for that would be great.

**Comment by haircommander** [comment]
failures reproduced, looking

**Comment by haircommander** [comment]
test fails when using a userns and I don't think it's the fault of this PR, so 'I've dropped it for now

**Comment by haircommander** [comment]
old failures do trigger here https://github.com/cri-o/cri-o/pull/5344

we'll stall on a test addition until that's figured out, but we should move forward here

File: server/container_create_linux.go [hunk 1]
```diff
@@ -458,13 +458,18 @@ func (s *Server) createSandboxContainer(ctx context.Context, ctr ctrIface.Contai
 		}
 
 		if !isInCRIMounts("/sys", containerConfig.Mounts) {
-			specgen.RemoveMount("/sys")
```
Comment (mrunalp) [inline server/container_create_linux.go:461]: What if we are privileged and host network?

Comment (haircommander) [reply server/container_create_linux.go:461]: the mount generating code is all over the place tbh. Right after this block, we overwrite the mount if we're privileged

Comment (mrunalp) [reply server/container_create_linux.go:461]: Can we skip the add if we are also privileged since they are added rw in that case?

**Comment by haircommander** [comment]
@wgahnagl @QiWang19 @fgiudici PTAL

**Comment by QiWang19** [comment]
LGTM

**Review by fgiudici — APPROVED** [review]
LGTM!

**Pull request merged**